controller/api/destination/watcher/ip_watcher.go

```diff
@@ -104,6 +104,15 @@ func (iw *IPWatcher) Unsubscribe(clusterIP string, port Port, listener EndpointU
 	ss.unsubscribe(port, listener)
 }
 
+// GetSvc returns the service that corresponds to an IP address if one exists.
+func (iw *IPWatcher) GetSvc(clusterIP string) (*ServiceID, bool) {
+	ss, exists := iw.getServiceSubscriptions(clusterIP)
```

Comment:
I think this will work because we create an serviceSubscription whenever there is a service added event.  However, it still feels a bit brittle and indirect.  A more direct approach would be to do something like `iw.k8sAPI.Svc().Informer().GetIndexer().ByIndex(podIPIndex, clusterIP)` to get the service directly.

Reply:
I see what you mean. I originally did it this way, but changed it to how it currently is since it ends up fairly similar to `getOrNewServiceSubscriptions`.

I think it's pretty safe to assume that the informers event handler will be triggered in a timely manner--and call `getOrNewServiceSubscriptions`, but see why this may be a more direct approach.

Reply:
This method has been changed since your review. On top checking the actual indexer, it now handles an IP that does _or_ does not map to a service and errors. The existence of a service that it maps to is used by the calling function to determine if it should subscribe to a service or not.